src/logic/vision.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2007-2020 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+
+#ifndef WL_LOGIC_VISION_H
+#define WL_LOGIC_VISION_H
+
+namespace Widelands {
+
+// Simplified vision information reduced to 3 states.
+// Do not change the order! It is stored in savegames.
+enum class VisibleState {
+	kUnexplored = 0,      // Never seen yet
+	kPreviouslySeen = 1,  // Previously seen
+	kVisible = 2          // Currently visible
+};
+
+// Extended vision information encoded as a number with some helper functions.
+// See Player::Field for explanation of values.
+struct Vision {
+	uint16_t value;
```

Comment:
Should this value be accessed directly? Better make it `private`, add a getter function, and either some functions or a `friend` for modifications.